edb/edgeql/desugar_group.py

```diff
@@ -199,7 +214,38 @@ def try_group_rewrite(
         )
     """
 
-    # TODO: would Python's new pattern matching fit well here???
+    # Inline trivial uses of aliases bound to a group and then
+    # immediately used, so that we can apply the other optimizations.
+    match node:
+        case qlast.SelectQuery(
+            aliases=[*_, qlast.AliasedExpr(
+                alias=alias, expr=qlast.GroupQuery() as grp)
+            ] as qaliases,
```

Comment:
Perhaps better formatted as

```suggestion
            aliases=[
                *_, 
                qlast.AliasedExpr(
                  alias=alias, 
                  expr=qlast.GroupQuery() as grp
                )
            ] as qaliases,
```

?